CODE_STYLE.md

```diff
@@ -0,0 +1,943 @@
+##To be added;
+
+returning errors
+documenting functions
+empty-tables
+
+
+# Mashape Kong contributing guidelines
+
+This guide contains a list of guidelines that we try to follow for the
+Kong codebase. 
+
+
+
+## <a name='TOC'>Table of Contents</a>
+
+  1. [Version control](#version-control)
+      - [Branches](#branches)
+      - [Commit message format](#commits)
+      - [Pull requests](#pullrequests)
+  1. [Testing](#testing)
+  1. [Performance](#performance)
+  1. [Lua Style Guide](#style-guide)
+      - [Tables](#tables)
+      - [Strings](#strings)
+      - [Functions](#functions)
+      - [Properties](#properties)
+      - [Variables](#variables)
+      - [constants](#constants)
+      - [Conditional Expressions & Equality](#conditionals)
+      - [Blocks](#blocks)
+      - [Whitespace](#whitespace)
+      - [Commas](#commas)
+      - [Semicolons](#semicolons)
+      - [Type Casting & Coercion](#type-coercion)
+      - [Naming Conventions](#naming-conventions)
+      - [Modules](#modules)
+
+## <a name='version-control'>Version control</a>
+
+### <a name='branches'>Branches</a>
+
+  There are two important branches in the repository:
+  
+  - `master` is the branch for stable, released code
+  - `next` a development branch where new features land
+  
+  When contributing the distinction is important. When contributing a bugfix found
+  in a release, base your fix of the `master` branch. New features and fixes on
+  `next` should be based of the `next` branch.
+
+  **[[back to top]](#TOC)**
+
+### <a name='commits'>Commit message format</a>
+
+  - This part of the guide started as a copy of the [Angular commit message format](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format)
+
+  - Please write insightful and descriptive commit messages. It lets us and future contributors
+    quickly understand your changes without having to read your changes. Please provide a
+    summary in the first line and eventually, go to greater lengths in
+    your message's body. We also like commit message with a **type** and **scope**.
+
+  - Please create commits containing **related changes**. For example, two different bugfixes
+    should produce two separate commits. A feature should be made of commits splitted by
+    **logical chunks** (no half-done changes). Use your best judgement as to how many commits
+    your changes require.
+
+  Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+  format that includes a **type**, a **scope** and a **subject**:
+
+  ```
+  <type>(<scope>): <subject>
+  <BLANK LINE>
+  <body>
```

Comment:
One thing that confused me at first was <> around BLANK LINE and around body - would it be more clear if all the <> were removed? Or, perhaps, if the <> around type, scope, subject, body, and footer were changed to []?